overlord/ifacestate/ifacestate_test.go

```diff
@@ -4962,7 +4962,7 @@ func (s *interfaceManagerSuite) TestAttributesRestoredFromConns(c *C) {
 	c.Check(conn.Slot.Attr("number", &number), IsNil)
 	c.Check(number, Equals, int64(1))
 
-	ifacestate.UpdateConnectionInConnState(conns, conn, false, false)
+	ifacestate.UpdateConnectionInConnState(conns, conn, false, false, false)
```

Comment:
(unreleated to this PR) I think this indicates that the API for UpdateConnectionInConnState can be improved, in order to see what the three "false" means one must go to the declaration of UpdateConnectionInConnState() to see what it is doing. Given that this is only in tests its not so critical and in the new code it looks fine, so maybe just change to something like:
```
var isAuto, byGadget, isUndesired bool
ifacestate.UpdateConnectionInConnState(conns, conn,  isAuto, byGadget, isUndesired)
```
here?

Reply:
Nice suggestion, could be used as an interim pattern for boolean values.

Reply:
Pushed